ansible/playbooks/AdoptOpenJDK_AIX_Playbook/main.yml

```diff
@@ -15,28 +15,39 @@
     swap_size: 4096
 
   roles:
+    # AIX base requirements
+    - X11
+    - openssl
+
+    # AIX configuration
     - aixfs
     - syslogs
-    - X11
+
+    # yum based installation of additional software
+    # installs some gnu packages required by ansible modules
+    - yum
```

Comment:
Despite your reasoning [here](https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1704#issuecomment-734921703), running the x11 role before yum resulted in 
```
fatal: [p9-aix1-ojdk06.osuosl.org]: FAILED! => {"changed": false, "msg": "Failed to find handler for \"/root/.ansible/tmp/ansible-tmp-1606750087.0092869-25003-46895319352335/source\". Make sure the required command to extract the file is installed. Command \"unzip\" not found. Command \"/usr/bin/tar\" detected as tar type None. GNU tar required."}
```

I then moved the yum role so that it would run first, and then ran the x11 role subsequently. Ansible did not have a problem unzipping and untarring the archive. The openssl role also includes a archive extract, so the yum role should run before this too.

Reply:
Just saw https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1695#discussion_r531706008

Out of curiosity, is there a technical reason for this order of installation? For example, will X11 install differently if it follows the yum and packages installation? And vice versa

Reply:
RPM packaging changes over time. Over the years there has been impact re: the order of installation. My expectation - today - is that it should not matter for most things.

Note: for 'yum'sh' to run, besides having to download it first (which I do not know, immediately, will work with just a base AIX system with no additional AIX software such as ftp to fetch the file) I do know that the script `yum.sh` does require the package `expect` which requires `tk` and `tcl` - plus it needs the `ftp` application - which on AIX 7.2 requires an additional fileset.

None of the requirements are visible when the installation is performed using `all`. IMHO - that is both insecure (installing things considered insecure by IBM since 1995 (`nis` is a good example) - but may also introduce unintended dependencies in the OJDK product - libraries get referenced not because they are really needed - but just because they are there and get included in dynamic load symbol lists.

When researching why things are broken - my quickest solution was to install on a minimal system and see what is demanded - and then verify that it is actually needed. This minimalist approach was not something I thought of on my own - but actually a early mentor over 30 years ago showed me the kinds of problems if prevented. The idiom: An ounce of prevention is better than a pound of cure". With this approach I have been able to resolve numerous issues others have looked at for months - with no explanation.

Back to you core question re: X11 and yum.sh - it should not matter - but I would rather not take the risk. Hence: install AIX bos first, and OSS additions later. If there are conflicts the package managers should report the conflicts. (note: should).

More specifically, the reason for the error you mention is the Ansible unarchive: module which assumes unzip. This assumption is not met by base AIX. Perhaps this message would not occur if the `vendor` file was not zipped and/or did not assume that gtar (which supports the `'z'` argument for uncompress versus using (for gzip compressed files) `gzip -dc <input> | tar xf -`

I am not expert enough to say Ansible has a bug. Rather I would say the design is not working on base AIX.

Reply:
Just to be clear - the yum role does run before the xlc ones. As yum, unzip, gtar, etc. are all 'external' to base AIX it really feels like putting that cart in front of the horse.

Again, I have no access to, nor do I want access to proprietary information - but is unzip even used? Or is it being demanded only because the Ansible unarchive module looks for it, regardless of need - just as it perhaps looks for gtar - when even AIX tar would work?